### PR TITLE
Scan: Fetch scan state with polling

### PR DESCRIPTION
**Pull request by ashiagr**
Parent: #13326

This PR 
- Adds use case to fetch scan state 
- Replaces fetching from `ScanStatusService` to newly added UseCase in the scan VM
- Copies progress state, VH from backup download implementation to common section for reuse in scan 
- Updates scan state builder to show progress item for SCANNING scan state

Start Scan Child PR: #13741

**To test**

Prerequisite: 

- Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
- WP.com account with a site having scan capability (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com).

1. Login to the app using above WP.com account and select the site on the My Site tab
2. Click scan menu item on My Site
3. Notice that scan state is fetched and shown
4. Return to My Site screen
5. Start scan for the site from the browser (Login to WP.com account on the web browser, select the site  -> Jetpack -> Scan, click "Scan now" button)
6. Click scan menu item on My Site again while scanning is in progress
7. Notice that scanning scan state is fetched and shown with a progress bar (progress updates shown in a label)
(Tests included in `FetchScanStateUseCaseTest` and `ScanViewModelTest` )

**Notes**

1. Ignore all styling
2. (Initial) Loading/ Failure states not handled
3. There's a small delay before UI is displayed on screen load (while first fetch is not complete). Showing last scan state for the site from the db (before first fetch) can be done either using a separate `GetScanStateUseCase` or from inside the `FetchScanStateUseCase`.

**Merge Instructions**

1. Review FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1833, merge and create a tag in FluxC develop
2. Replace FluxC hash with tag from develop 
2. Remove the "Not Ready for Merge label"
3. Merge this PR

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanStateUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            return@flow
+        } else {
+            val scanStateModel = scanStore.getScanStateForSite(site)
+            if (scanStateModel != null) {
+                emit(Success(scanStateModel))
+                if (polling && scanStateModel.state == ScanStateModel.State.SCANNING) {
+                    fetchScanState(site = site, polling = true).collect {
+                        emit(it)
+                        delay(delayInMs)
+                    }
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanStateUseCase.kt:49]: This is a tricky part with a recursive call to the `fetchScanState` function during polling. We might want to replace it with a `while` loop.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt [hunk 2]
```diff
@@ -38,12 +44,20 @@ class ScanViewModel @Inject constructor(
...
+        fetchScanState()
     }
 
-    private fun init() {
-        scanStatusService.scanState.observeForever(scanStateObserver)
-        scanStatusService.start(site)
+    private fun fetchScanState() {
+        launch {
+            fetchScanStateUseCase.fetchScanState(site = site)
+                .flowOn(bgDispatcher)
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt]: 🔍 Nitpick: We might want to consider setting the dispatcher in the usecase so no-one can invoke the usecase on the ui thread. Wdyt?

**Review by malinajirka — COMMENTED** [review]
Thanks @ashiagr! I reviewed the changes and it looks great to me. However, I didn't time to finish testing the behavior. I'll wrap it up and merge the PR first thing in the morning.

**Review by malinajirka — APPROVED** [review]
I've tested the app and it all works as described, good job! ⭐

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt (hunk 2 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt]: Nice suggestion 👍. 
In 79b6480

**Review by malinajirka — APPROVED** [review]
Thanks @ashiagr! LGTM - feel free to merge it when you replace the fluxc hash.

**Pull request merged**